lib/components/base/Checkbox.vue

```diff
@@ -25,43 +25,38 @@
 </template>
 <script setup>
 import { CheckIcon, DropdownIcon } from '@'
-</script>
-<script>
-import { defineComponent } from 'vue'
-
-export default defineComponent({
-  props: {
-    label: {
-      type: String,
-      default: '',
-    },
-    disabled: {
-      type: Boolean,
-      default: false,
-    },
-    description: {
-      type: String,
-      default: '',
-    },
-    modelValue: Boolean,
-    clickEvent: {
-      type: Function,
-      default: () => {},
-    },
-    collapsingToggleStyle: {
-      type: Boolean,
-      default: false,
-    },
+
+const emit = defineEmits(['update:modelValue'])
+
+const props = defineProps({
+  label: {
+    type: String,
+    default: '',
   },
-  emits: ['update:modelValue'],
-  methods: {
-    toggle() {
-      if (!this.disabled) {
-        this.$emit('update:modelValue', !this.modelValue)
-      }
-    },
+  disabled: {
+    type: Boolean,
+    default: false,
+  },
+  description: {
+    type: String,
+    default: '',
+  },
+  modelValue: Boolean,
+  clickEvent: {
+    type: Function,
+    default: () => {},
+  },
+  collapsingToggleStyle: {
+    type: Boolean,
+    default: false,
   },
 })
```

Comment:
Just like the other PR, this would be cleaner with TypeScript generic and `withDefaults`.

Reply:
Done ^^.